crates/storage/provider/src/traits/chain.rs

```diff
@@ -139,3 +139,44 @@ impl CanonStateNotification {
         receipts
     }
 }
+
+/// Wrapper around a broadcast receiver that receives fork choice notifications.
+#[derive(Debug)]
```

Comment:
can we please derive `derive_more::Deref` and `derive_more::DerefMut` for wrapper types